frontend/app/routes/settings/account-settings.tsx

```diff
@@ -1,14 +1,173 @@
+        <h2 className="text-2xl">Profile</h2>
+      </div>
+      <Separator />
+      <p className="text-grey">Update your profile information</p>
+      <div className="grid lg:grid-cols-12 gap-10 relative">
+        <div className="lg:col-span-10 flex flex-col">
+          <section id="update-profile" className="flex gap-1 scroll-mt-20">
+            <div className="w-16 hidden md:flex flex-col items-center">
+              <div className="flex rounded-full size-10 flex-none items-center justify-center p-1 border-2 border-grey/50">
+                <UserIcon size={15} className="flex-none text-grey" />
+              </div>
+              <div className="h-full border border-grey/50"></div>
+            </div>
+            <div className="w-full flex flex-col gap-5 pt-1 pb-8">
+              <h3 className="text-lg text-grey">Profile Information</h3>
+              <UpdateProfileForm />
+            </div>
+          </section>
+
+          <section id="update-password" className="flex gap-1 scroll-mt-20">
+            <div className="w-16 hidden md:flex flex-col items-center">
+              <div className="flex rounded-full size-10 flex-none items-center justify-center p-1 border-2 border-grey/50">
+                <KeyIcon size={15} className="flex-none text-grey" />
+              </div>
+            </div>
+            <div className="w-full pt-1 pb-8 flex flex-col gap-2">
+              <h3 className="text-lg text-grey">Change Password</h3>
+              <div>
+                <Link
+                  to="/settings/account/change-password"
+                  className="hover:underline text-sm py-2 flex items-center gap-0.5 text-link"
+                >
+                  <span>Change your password here</span>
+                  <ChevronRight size={15} />
+                </Link>
+              </div>
+            </div>
+          </section>
+        </div>
+      </div>
+    </section>
+  );
+}
+
+function UpdateProfileForm() {
+  const fetcher = useFetcher<typeof clientAction>();
+  const { data: userData } = useQuery(userQueries.authedUser);
+
+  const isPending = fetcher.state !== "idle";
+  const errors = getFormErrorsFromResponseData(fetcher.data?.errors);
+
+  const user = userData?.data?.user;
+
+  return (
+    <fetcher.Form method="POST" className="flex flex-col gap-6">
+      {errors.non_field_errors && (
+        <Alert variant="destructive">
+          <AlertCircle className="h-4 w-4" />
+          <AlertTitle>Error</AlertTitle>
+          <AlertDescription>{errors.non_field_errors}</AlertDescription>
+        </Alert>
+      )}
+
+      <div className="flex flex-col gap-4">
+        <FieldSet
+          name="username"
+          required
+          errors={errors.username}
+          className="flex flex-col gap-2"
+        >
+          <FieldSetLabel className="block">Username</FieldSetLabel>
+          <FieldSetInput
+            placeholder="Enter your username"
+            defaultValue={user?.username || ""}
```

Comment:
```suggestion
            defaultValue={user.username}
```

Since `user` will not be `undefined`, no need to use `?` also, even if it was `undefined`, you could simply use `user?.username`, `defaultValue` also accept `undefined`